### PR TITLE
CMake: fix build race condition

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: runtime/flang/CMakeLists.txt [hunk 1]
```diff
@@ -475,15 +475,23 @@ SET(FTN_SUPPORT
...
   ieee_exceptions.F95
-  iso_c_bind.F95
-  )
+)
 
 add_flang_library(flang_static
   ${FTN_INTRINSICS}
   ${FTN_SUPPORT}
   ${SHARED_SOURCES}
+  $<TARGET_OBJECTS:iso_c_bind>
```
Comment (isuruf) [inline runtime/flang/CMakeLists.txt:493]: This is wrong. These objects are built with different compile options than the library.

**Comment by isuruf** [comment]
I think we should go back to https://github.com/isuruf/flang/pull/102 where it was working fine.

**Comment by ghost** [comment]
It may be possible that Ninja will detect the dependencies correctly now that the module directory is set.

**Comment by ghost** [comment]
Note that gh-102 had a race condition as well that was concealed by the conditional statement: adding library dependencies doesn't work when:

1. The same object is compiled twice
2. The module directories are different

**Comment by isuruf** [comment]
https://github.com/isuruf/flang/tree/windows2

**Pull request merged**